src/Operation/Find.php

```diff
@@ -313,6 +323,10 @@ public function execute(Server $server)
 
         $cursor = $server->executeQuery($this->databaseName . '.' . $this->collectionName, new Query($this->filter, $this->createQueryOptions()), $this->createExecuteOptions());
 
+        if (isset($this->options['codec'])) {
+            return CodecCursor::fromCursor($cursor, $this->options['codec']);
+        }
+
         if (isset($this->options['typeMap'])) {
```

Comment:
The PR description says:

> When specifying a codec, the typeMap option is ignored entirely, and a bson type is applied for the root element.

Does it make sense to throw if both options are specified? I wouldn't consider that a BC break since we're introducing a new option anyway. Seems like it would be a safe way to avoid users doing something incorrect (as we do for some other operations by warning about conflicting options).

Also, I noticed there are no docs in this PR. Did you intend to lump those changes together or handle it later? If later, please add `documentation-needed` to PHPLIB-1182.

Reply:
I wasn't sure whether to lump the docs in here or not. I've created PHPLIB-1209 to track this explicitly.

Reply:
What about prohibiting `codec` and `typeMap`?

Reply:
I added logic to prohibit providing both a type map and a non-null codec. I also used the oppoertunity to de-duplicate some of the inheritance logic in the Collection class.